pydicom/pixel_data_handlers/rle_handler.py

```diff
@@ -0,0 +1,171 @@
+        format_str = 'uint{}'.format(dicom_dataset.BitsAllocated)
+    elif dicom_dataset.PixelRepresentation == 1:
+        format_str = 'int{}'.format(dicom_dataset.BitsAllocated)
+    else:
+        format_str = 'bad_pixel_representation'
+    try:
+        numpy_format = numpy.dtype(format_str)
+    except TypeError:
+        msg = ("Data type not understood by NumPy: "
+               "format='{}', PixelRepresentation={}, "
+               "BitsAllocated={}".format(
+                   format_str,
+                   dicom_dataset.PixelRepresentation,
+                   dicom_dataset.BitsAllocated))
+        raise TypeError(msg)
+
+    print("Decoding RLE")
+    UncompressedPixelData = bytearray()
+
+    if ('NumberOfFrames' in dicom_dataset and
+            dicom_dataset.NumberOfFrames > 1):
+
+        CompressedPixelDataSeq = pydicom.encaps.decode_data_sequence(
+            dicom_dataset.PixelData)
+
+        for frame in CompressedPixelDataSeq:
+            decompressed_frame = rle_decode_frame(frame,
+                                                  Rows=dicom_dataset.Rows,
+                                                  Columns=dicom_dataset.Columns,
+                                                  SamplesPerPixel=dicom_dataset.SamplesPerPixel,
+                                                  BitsAllocated=dicom_dataset.BitsAllocated)
+
+            UncompressedPixelData.extend(decompressed_frame)
+
+    else:
+
+        CompressedPixelData = pydicom.encaps.defragment_data(
+            dicom_dataset.PixelData)
+
+        decompressed_frame = rle_decode_frame(CompressedPixelData,
+                                              Rows=dicom_dataset.Rows,
+                                              Columns=dicom_dataset.Columns,
+                                              SamplesPerPixel=dicom_dataset.SamplesPerPixel,
+                                              BitsAllocated=dicom_dataset.BitsAllocated)
+
+        UncompressedPixelData.extend(decompressed_frame)
+
+    pixel_array = numpy.frombuffer(UncompressedPixelData, numpy_format)
+
+    return pixel_array
+
+
+def rle_decode_frame(d, Rows, Columns, SamplesPerPixel, BitsAllocated):
+    rle_start = 0
+    rle_len = len(d)
+
+    number_of_planes = unpack(b'<L', d[rle_start: rle_start + 4])[0]
+
+    if BitsAllocated % 8 != 0:
+        raise Exception("Don't know how to handle BitsAllocated not being a multiple of bytes")
+
+    BytesAllocated = BitsAllocated // 8
+
+    expected_number_of_planes = SamplesPerPixel * BytesAllocated
+
+    if number_of_planes != expected_number_of_planes:
+        raise Exception("Unexpected number of planes")
+
+    plane_start_list = []
+    for i in range(number_of_planes):
+        plane_start_in_rle = unpack(b'<L', d[rle_start + 4 + (4 * i):rle_start + 4 + (4 * (i + 1))])[0]
+        plane_start_list.append(plane_start_in_rle + rle_start)
+
+    plane_end_list = plane_start_list[1:]
+    plane_end_list.append(rle_len + rle_start)
```

Comment:
Are you sure - It works with 2+6 planes on my test files?
plane_end_list is the end of each plane so the start of the planes shifted down by one with the total length tacked on at the end